klippa/src/glyf_loca.rs

```diff
@@ -121,12 +184,39 @@ fn subset_simple_glyph(g: &SimpleGlyph) -> Vec<u8> {
     if i == 0 {
         return out;
     }
-    let trimmed_len =
-        10 + 2 * (g.number_of_contours() as usize) + 2 + g.instruction_length() as usize + i;
-    let Some(trimmed_slice) = g.offset_data().as_bytes().get(0..trimmed_len) else {
+
+    let glyph_bytes = g.offset_data().as_bytes();
+    let header_len = 10 + 2 * (g.number_of_contours() as usize) + 2;
+    let Some(header_slice) = glyph_bytes.get(0..header_len) else {
         return out;
     };
+    out.extend_from_slice(header_slice);
+
+    if plan
+        .subset_flags
+        .contains(SubsetFlags::SUBSET_FLAGS_NO_HINTING)
+    {
+        out[header_len - 2] = 0;
+        out[header_len - 1] = 0;
+    } else {
+        let instruction_end = header_len + g.instruction_length() as usize;
+        let Some(instruction_slice) = glyph_bytes.get(header_len..instruction_end) else {
+            return Vec::new();
+        };
+        out.extend_from_slice(instruction_slice);
+    }
+
+    let Some(trimmed_slice) = glyph_data.get(0..i) else {
+        return Vec::new();
+    };
```

Comment:
again I don't think this is reachable if we got this far? But it does raise the question of whether or not we want to be reporting errors...

Reply:
It might be reachable if something wrong in calculating trimmed length